whisper.cpp

```diff
@@ -179,6 +207,7 @@ static struct ggml_tensor * ggml_mul_mat_pad(struct ggml_context * ctx, struct g
 }
 
 // TODO: check if other platforms can benefit from this optimization
+// TODO: CUDA is currently broken - seems ggml_mul_mat does not handle views correctly
 #if defined(GGML_USE_METAL)
 #define ggml_mul_mat ggml_mul_mat_pad
 #endif
```

Comment:
The `ggml_mul_mat_pad` trick is very useful for the Metal kernels and provides significant improvement for the encoder.

Currently, this trick does not work with CUDA because we seem to have issues in some cases when the `src` are non-contiguous views. At the very least `ggml_cuda_mul_mat_mat_batched_cublas` does not handle all cases correctly for `src1` being non-contiguous because `ggml_get_to_fp16_cuda()` assumes contiguous data, but there might be other issues as well.

We should keep this in mind and fix or assert properly